widget/cardcontainer.go

```diff
@@ -0,0 +1,211 @@
+package widget
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/theme"
+)
+
+// CardContainer widget groups title, subtitle with content and a header image
+type CardContainer struct {
+	BaseWidget
+	Title, SubTitle string
+	Image           *canvas.Image
+	Content         fyne.CanvasObject
+}
+
+// NewCardContainer creates a new card widget with the specified title, subtitle and content (all optional).
+func NewCardContainer(title, subtitle string, content fyne.CanvasObject) *CardContainer {
+	card := &CardContainer{
+		Title:    title,
+		SubTitle: subtitle,
+		Content:  content,
+	}
+
+	card.ExtendBaseWidget(card)
+	return card
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (c *CardContainer) CreateRenderer() fyne.WidgetRenderer {
+	c.ExtendBaseWidget(c)
+
+	header := canvas.NewText(c.Title, theme.TextColor())
+	header.TextStyle.Bold = true
+	subHeader := canvas.NewText(c.SubTitle, theme.TextColor())
+
+	objects := []fyne.CanvasObject{header, subHeader}
+	if c.Image != nil {
+		objects = append(objects, c.Image)
+	}
+	if c.Content != nil {
+		objects = append(objects, c.Content)
+	}
```

Comment:
If Image or Content are set after a delay (eg goroutine downloads image from network, goroutine loads content from file), is it possible the widget's renderer is created before the new Image and Content are set and so they will not be in the renderer's objects, thus not rendered?

Reply:
Updated and 2 new tests demonstrate it